### PR TITLE
fix(scripts/RuinsOfAhnQiraj): Improve Kurinnaxx

### PR DESCRIPTION
**Pull request by Si1ker**
Originated from #12061

## Changes Proposed:
- Cleanup/refactor
- Migrate trash spell handling from script to db
- Guides state that cleave (WideSlash) applies the 'Mortal Wound' debuff but videos demonstrate that it does not apply them. **Those are separate events and separate spells**
source: https://youtu.be/EgazIWGM0n0?t=30 AND https://youtu.be/idACansl4-0?t=3 / https://youtu.be/idACansl4-0?t=14
- Adjust timers based on [Mangos](https://github.com/mangos/ScriptDev3/blob/master/scripts/kalimdor/ruins_of_ahnqiraj/boss_kurinnaxx.cpp)

## Tests Performed:

- Tested

## How to Test the Changes:

1. Go to AQ20
2. Engage Kurinaxx encounter

## Known Issues and TODO List:

* Sand traps require scripting.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Si1ker** [comment]
ready

**Review by acidmanifesto — DISMISSED** [review]
Now I really do like the idea of DoCastSelf in place of Docast(me,XXX) as that is more appropiate. But i am curious why are we moving thrash to a creature aura in the addons? You made mention of referencing mangos for the adjustment due to their timer, where they also have thrash on a timer as well. What was the determining factor in your choice in moving it to the db is not exactly clear to me, can you provide some insight in that decision? We had a past incident with a massive thrash pr that had to be clean up manually and just want to make sure we are not ending up in a repeat with this creature atleast.

**Review by Nyeriah — APPROVED** [review]

**Review by Nyeriah — CHANGES_REQUESTED** [review]

File: src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_kurinnaxx.cpp [hunk 1]
```diff
@@ -87,6 +84,11 @@ class boss_kurinnaxx : public CreatureScript
 
             events.Update(diff);
 
+            if (me->GetDistance(me->GetVictim()) > 60.00f && me->GetDistance(me->GetVictim()) < 120.00f)
+            {
+                DoCastVictim(SPELL_SUMMON_PLAYER);
+            }
+
```
Comment (Nefertumm) [inline src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_kurinnaxx.cpp:91]: This needs to be double-checked. This spell is usually used by boss to summon unreachable players.

**Comment by Gultask** [comment]
Tested.

There is the usual inconsistency with Thrash, as it can trigger 3 times in a row and then not trigger for the next 30s. But that's expected with the aura.

The Mortal Wounds and Wide Slash events work.

One question. Would Nix PR work for this boss? So that it won't need the summon player script? Or is it different and based on distance and not due to evades? The arena is rather complex.

**Comment by Nefertumm** [comment]
@Gultask indeed, that's why I wrote that it needs to be double checked. With Nix' PR it won't be needed on this PR.

File: data/sql/updates/pending_db_world/rev_1655863615360060800.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+--
+DELETE FROM `creature_addon` WHERE `guid`=144632;
```
[the remark this answers is not in the document]
Comment (Si1ker) [reply data/sql/updates/pending_db_world/rev_1655863615360060800.sql:2]: we've been using creature_addon, so will use it.

**Comment by Gultask** [comment]
I took a quick look at the boss in retail and the TP mechanic was present there, albeit broken since it was spammed so hard I could not move away from the boss. I'll try it again someday with another class to see if there's a difference.

And this is the PR I was talking about: https://github.com/azerothcore/azerothcore-wotlk/pull/12193

**Review by Nefertumm — APPROVED** [review]

**Comment by Si1ker** [comment]
## Known Issues and TODO List:

* Sand traps require scripting.

**Pull request merged**